utils/sv_bmesh_utils.py

```diff
@@ -162,7 +162,12 @@ def add_mesh_to_bmesh(bm, verts, edges=None, faces=None, sv_index_name=None, upd
         new_edges = []
     #[new_face([bm_verts[i] for i in face]) for face in faces or []] # не надо так делать
     if faces is not None:
-        for face in faces:
+        # remove double faces. Some times get faces with same verts but with another order: [62,103,102] and [102,103,62].
+        # this is not compatible with mesh and only one of that faces can be used. So convert this faces indices into dictionary.
+        # this remove faces with the same vertices but same or different orders. This may happens on some of Voronoi if
+        # it slice self intersected mesh.
+        dict_faces = { tuple(sorted(f)): f for f in list(faces) }
```

Comment:
nice job

Reply:
I'm interesting how much it effects performance?

Reply:
Hi @Durman . I think it is not too much:
![image](https://github.com/nortikin/sverchok/assets/14288520/fb5adb94-c226-4896-b4e2-b0de2209b586)

Is a some trick to optimize it?

Reply:
I wouldn't measure the performance in this way. I mean it would be nice know how much present the node is slower, especially in cases where this extra line does nothing useful. For example: before it took 1 second now it takes 1,1 second to handle the same mesh by the node.

Reply:
If exclude this line then next line will raise exceptions in case of creating face with the same vertices.